app/platform/usb/hid.h

```diff
@@ -19,4 +19,4 @@
 
 err_t usb_hid_recv(struct usb_rx_queue_item *p_rx_queue_item, uint32_t timeout_ticks);
 err_t usb_hid_send(uint8_t *p_buf, uint16_t len);
-err_t usb_hid_init(USBD_HandleTypeDef *p_usbd, PCD_HandleTypeDef *p_pcd);
+err_t usb_hid_init(const void *p_data);
```

Comment:
perhaps use a target-specific typedef'ed value instead? e.g. `const platform_usb_pdata_t *p_data;`
with that as
`typedef USBD_HandleTypeDef platform_usb_pdata_t;`

Reply:
Mm like the gpio_impl.h I suppose, so each platform gets its own init function prototype.